### DISCUSSION TITLE
DataPortal no longer can find a "good enough" method match in ObjectFactory when method exists in base class.

### DISCUSSION THREAD

**Discussion by kellyethridge** [MarimerLLC/csla#1705] [Questions] [answered]
Hello,

If I have a `Fetch` method defined as `Fetch(object criteria)` in an `ObjectFactory` base class. CSLA no longer discovers it as an acceptable match. However, if I implement the same method in the derived factory class CSLA works as expected. This used to work without issue in CSLA 4.

Is this the expected behavior? It seems like it shouldn't be since this works fine in a business class using `DataPortal_Fetch(object criteria)` method signature even in a base class.

Edit 1: I can't get the `Fetch` method in a base `ObjectFactory` class to work even if the criteria is an exact match. It fails with a `TargetParameterCountException`. There is only a single Int32 being passed.

Thank you.

**Comment by rockfordlhotka** [comment, marked as answer]
The object factory model is unchanged from earlier versions of CSLA - I've implemented the new capabilities only for the encapsulated models so far.
